execute/plugin_usdc_test.go

```diff
@@ -0,0 +1,515 @@
+		return m.Header.SequenceNumber
+	})
+	require.ElementsMatch(t, sequenceNumbers, []cciptypes.SeqNum{102, 103, 104})
+	//Attestation data added to the USDC
+	require.NotEmpty(t, outcome.Report.ChainReports[0].OffchainTokenData[2])
+
+	server.AddResponse(
+		"0x70ef528624085241badbff913575c0ab50241e7cb6db183a5614922ab0bcba5d",
+		`{
+						"status": "complete",
+						"attestation": "0x720502893578a89a8a87982982ef781c18b194"
+					}`)
+
+	// Run 3 more rounds to get all attestations
+	res, err = runner.RunRound(ctx)
+	require.NoError(t, err)
+	res, err = runner.RunRound(ctx)
+	require.NoError(t, err)
+	res, err = runner.RunRound(ctx)
+	require.NoError(t, err)
+	outcome, err = exectypes.DecodeOutcome(res.Outcome)
+	require.NoError(t, err)
+	sequenceNumbers = slicelib.Map(outcome.Report.ChainReports[0].Messages, func(m cciptypes.Message) cciptypes.SeqNum {
+		return m.Header.SequenceNumber
+	})
+	require.ElementsMatch(t, sequenceNumbers, []cciptypes.SeqNum{102, 103, 104, 105})
+	//Attestation data added to the both USDC messages
+	require.NotEmpty(t, outcome.Report.ChainReports[0].OffchainTokenData[2])
+	require.NotEmpty(t, outcome.Report.ChainReports[0].OffchainTokenData[3])
+}
+
+type nodeSetup struct {
+	node            *execute.Plugin
+	reportCodec     cciptypes.ExecutePluginCodec
+	msgHasher       cciptypes.MessageHasher
+	TokenDataReader *mock_types.MockTokenDataReader
+}
+
+func setupHomeChainPoller(
+	t *testing.T,
+	donID plugintypes.DonID,
+	lggr logger.Logger,
+	chainConfigInfos []reader.ChainConfigInfo,
+) reader.HomeChain {
+	const ccipConfigAddress = "0xCCIPConfigFakeAddress"
+
+	homeChainReader := readermock.NewMockContractReaderFacade(t)
+	var firstCall = true
+	homeChainReader.On(
+		"GetLatestValue",
+		mock.Anything,
+		mock.Anything,
+		mock.Anything,
+		mock.MatchedBy(func(input map[string]interface{}) bool {
+			_, pageIndexExists := input["pageIndex"]
+			_, pageSizeExists := input["pageSize"]
+			return pageIndexExists && pageSizeExists
+		}),
+		mock.Anything,
+	).Run(
+		func(args mock.Arguments) {
+			arg := args.Get(4).(*[]reader.ChainConfigInfo)
+			if firstCall {
+				*arg = chainConfigInfos
+				firstCall = false
+			} else {
+				*arg = []reader.ChainConfigInfo{} // return empty for other pages
+			}
+		}).Return(nil)
+
+	homeChainReader.EXPECT().
+		GetLatestValue(mock.Anything, types.BoundContract{
+			Address: ccipConfigAddress,
+			Name:    consts.ContractNameCCIPConfig,
+		}.ReadIdentifier(consts.MethodNameGetOCRConfig), primitives.Unconfirmed, map[string]any{
+			"donId":      donID,
+			"pluginType": consts.PluginTypeExecute,
+		}, mock.Anything).
+		Run(
+			func(
+				ctx context.Context,
+				readIdentifier string,
+				confidenceLevel primitives.ConfidenceLevel,
+				params,
+				returnVal interface{},
+			) {
+				*returnVal.(*[]reader.OCR3ConfigWithMeta) = []reader.OCR3ConfigWithMeta{{}}
+			}).
+		Return(nil)
+
+	homeChain := reader.NewHomeChainConfigPoller(
```

Comment:
I know this isn't part of your PR but do we need to run the real home chain poller instead of just a mock?